aztk/client/base/helpers/get_application_log.py

```diff
@@ -0,0 +1,114 @@
+import time
+
+import azure
+import azure.batch.models as batch_models
+import azure.batch.models.batch_error as batch_error
+
+from aztk import error
+from aztk import models
+from aztk.utils import constants, helpers
+
+output_file = constants.TASK_WORKING_DIR + \
+    "/" + constants.SPARK_SUBMIT_LOGS_FILE
+
+
+def __check_task_node_exist(batch_client, cluster_id: str, task: batch_models.CloudTask) -> bool:
+    try:
+        batch_client.compute_node.get(cluster_id, task.node_info.node_id)
+        return True
+    except batch_error.BatchErrorException:
+        return False
+
+
+def __wait_for_app_to_be_running(batch_client, cluster_id: str, application_name: str) -> batch_models.CloudTask:
+    """
+        Wait for the batch task to leave the waiting state into running(or completed if it was fast enough)
+    """
+    while True:
+        task = batch_client.task.get(cluster_id, application_name)
+
+        if task.state is batch_models.TaskState.active or task.state is batch_models.TaskState.preparing:
+            # TODO: log
```

Comment:
todo?

Reply:
This is just reshuffled code I believe. We don't currently have a logger in the SDK (we should add one though) so this is outside the scope for this PR I think.